src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/XElementExtensions.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace Microsoft.AspNetCore.Server.IntegrationTesting.IIS
+{
+    internal static class XElementExtensions
+    {
+        public static XElement RequiredElement(this XElement element, string name)
+        {
+            var existing = element.Element(name);
+            if (existing == null)
+            {
+                throw new InvalidOperationException($"Element with name {name} not found in {element}");
+            }
+
+            return existing;
+        }
+
+        public static XElement AddOrUpdate(this XElement element, string name)
```

Comment:
Nit: this doesn't update the existing element.

Reply:
GetOrAdd?